gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_clusters.hpp

```diff
@@ -60,14 +61,17 @@ pcl::gpu::extractEuclideanClusters (const pcl::PointCloud<pcl::PointXYZ>::Ptr  &
     max_answers = host_cloud_->size();
   else
     max_answers = max_pts_per_cluster;
-  std::cout << "Max_answers: " << max_answers << std::endl;
+  PCL_DEBUG("Max_answers: %i\n", max_answers);
 
   // to store the current cluster
   pcl::PointIndices r;
 
   DeviceArray<PointXYZ> queries_device_buffer;
   queries_device_buffer.create(max_answers);
 
+  // Host buffer for results
+  std::vector<int> sizes, data;
+
```

Comment:
Was the intention to remove the preallocations again?

Reply:
We were not sure about the correct size and realized that virtually all the performance gains can be realized by just keeping the vectors persistently in scope and let the resize function find the correct capacity.